code/modules/cm_aliens/hivebuffs/hivebuff.dm

```diff
@@ -0,0 +1,517 @@
+/*****************************
+*
+*	HIVE BUFFS	- XENOMORPH ENDGAME
+*	Contains all the class definitons and code for applying hivebuffs to xeno hives.
+*	Each buff consists of a /datum/hivebuff
+*	And associated on_engage and on_cease procs to handle behaviour of activating and ending the buffs
+*	Buffs are divided into 2 tiers, minor and major.
+*
+*****************************/
+
+///GLOBAL DEFINES///
+
+#define HIVE_STARTING_BUFFPOINTS 0
+#define BUFF_POINTS_NAME "Royal resin"
+
+///LOCAL DEFINES///
+
+#define HIVEBUFF_TIER_MINOR "Minor"
+#define HIVEBUFF_TIER_MAJOR "Major"
+
+
+/////////////////////////////
+
+/datum/hivebuff
+	/// Timer id to call on_cease() if neccessary to end the effects.
+	var/_timer_id
+	/// The hive that this buff is applied to.
+	var/datum/hive_status/hive
+	/// List of pylons sustaining the hive buff, can be one or two pylons
+	var/list/sustained_pylons
+	///Name of the buff, short and to the point
+	var/name = "Hivebuff"
+	/// Description of what the buff does.
+	var/desc = "Base hivebuff"
+	/// Path to .dmi with hivebuff radial icons
+	var/hivebuff_radial_dmi = 'icons/ui_icons/hivebuff_radial.dmi'
+	/// Image to display on radial menu
+	var/radial_icon = "health"
+	/// Round time before the buff becomes available to purchase
+	var/roundtime_to_enable = 0 HOURS
+
+	/// Flavour message to announce to the hive on buff application. Narrated to all players in the hive.
+	var/engage_flavourmessage = "The Queen has purchased a buff!"
+	/// Flavour message to announce to the hive on buff expiry. Narrated to all players in the hive.
+	var/cease_flavourmessage = "The buff has expired."
+
+	/// Minor or Major buff. Governs announcements made and importance.
+	var/tier = HIVEBUFF_TIER_MINOR
+	/// Number of pylons required to buy the buff
+	var/number_of_required_pylons = 1
+	///If this buff can be used with others
+	var/is_unique = TRUE
```

Comment:
Please refactor this to `is_combineable` or `is_mixable` because how its being used doesn't make sense. Also there's no buff that changes this (destroyer just changes it to the default value again so no change). So likely it should default to false, and destroyer sets it true if the goal was to only have 1 buff + destroyer?